assembly/src/ast/parsers/debug.rs

```diff
@@ -29,6 +29,38 @@ pub fn parse_debug(op: &Token) -> Result<Node, ParsingError> {
             }
             _ => return Err(ParsingError::extra_param(op)),
         },
+        "mem" => match op.num_parts() {
+            2 => DebugOptions::MemAll,
+            3 => {
+                let n: u32 = parse_checked_param(op, 2, 1..=u32::MAX)?;
+                DebugOptions::MemInterval(n, n)
+            }
+            4 => {
+                let n: u32 = parse_checked_param(op, 2, 0..=u32::MAX)?;
+                let m: u32 = parse_checked_param(op, 3, 0..=u32::MAX)?;
+                if m < n {
+                    return Err(ParsingError::invalid_param_with_reason(op, 3, "the index of the end of the interval must be greater than the index of its beginning"));
+                }
+                DebugOptions::MemInterval(n, m)
+            }
+            _ => return Err(ParsingError::extra_param(op)),
+        },
+        "local" => match op.num_parts() {
+            2 => DebugOptions::LocalInterval(0, 2u32.pow(16), 0),
+            3 => {
+                let n: u32 = parse_checked_param(op, 2, 0..=u16::MAX as u32)?;
+                DebugOptions::LocalInterval(n, n, 0)
+            }
+            4 => {
+                let n: u32 = parse_checked_param(op, 2, 0..=u16::MAX as u32)?;
+                let m: u32 = parse_checked_param(op, 3, 0..=u16::MAX as u32)?;
+                if m < n {
+                    return Err(ParsingError::invalid_param_with_reason(op, 3, "the index of the end of the interval must be greater than the index of its beginning"));
+                }
+                DebugOptions::LocalInterval(n, m, 0)
```

Comment:
Rather than setting number of local constants to 0 here and then updating later, we can probably pass the number of local constants into this function (similar to how we do it for `parse_loc_*` functions.

Reply:
I'm not sure I understand what do you mean. As far as I know `local_constants` contains only constants declared with `CONST` word, so number of locals is not stored there. It is stored in the `local_procs: LocalProcMap` (in `ProcedureAst` [here](https://github.com/0xPolygonMiden/miden-vm/blob/775209f1dd0b3903a870f22cb2b804347697d439/assembly/src/ast/mod.rs#L740)), but this information becomes available only after the procedure is completely parsed, so we can't get `num_locals` being within parsed procedure. That's why I had to create this ugly solution with updating already created `DebugOptions::LocalInterval` at the compilation stage, when we already have this data in the AssemblyContext. 

I hope that I just didn't understand something and there is a way to get number of locals at the parsing stage to avoid modifying `DebugOptions` later.

Reply:
Ah, sorry! You are correct - we currently don't have this info. But looking at the code again, it actually doesn't seem like it should be too difficult to track it.

What we can do is add `num_proc_locals: Option<u16>` to the [ParserContext](https://github.com/0xPolygonMiden/miden-vm/blob/main/assembly/src/ast/parsers/context.rs#L13). Then, in [parse_procedure()](https://github.com/0xPolygonMiden/miden-vm/blob/main/assembly/src/ast/parsers/context.rs#L292) you could set it to the number of locals right before we parse the procedure body, and then reset it back to `None` after procedure body is parsed.

We can probably use this info in other places as well - but that's for a different PR.